bridge-forwarder/src/xcm_asset_transactor.rs

```diff
@@ -27,9 +27,9 @@ impl<
 	// 2. recipient is on non-substrate chain(evm, cosmos, etc.)
 	// 3. recipient is on the remote parachain
 	fn deposit_asset(what: &MultiAsset, who: &MultiLocation, context: &XcmContext) -> XcmResult {
-		match (who.parents, who.first_interior()) {
+		match (who.parents, who.interior) {
 			// 1. recipient is on the local parachain
-			(0, Some(AccountId32 { .. })) | (0, Some(AccountKey20 { .. })) => {
+			(0, X1(AccountId32 { .. })) | (0, X1(AccountKey20 { .. })) | (1, X1(Parachain(_))) => {
```

Comment:
Any case that deposits to location `(1, X1(Parachain_))`? In my understanding, it seems like forward to  sibling parachian, but the `interior` length should be at least 2 that make sense

Reply:
this is the case that I was asking you on slack that raised `ExceedsStackLimit` error. When I was testing the case `sygma -> bridge hub -> asset hub` for Asset hub native token, the recipient `(1, {X2(parachain: 1000, accountID:xxxxx)})` got parsed correctly for the fist time it went through `deposit_asset` and then the logic went into xcm instruction `TransferReserveAsset` in [xcm-executor](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-executor/src/lib.rs#L635), and then it tries to call `Self::deposit_asset` in [transact_asset](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/xcm/xcm-executor/src/traits/transact_asset.rs#L134) again which cause the loop logic here, however, the implementation here was to match the first interior which will always be matched so it goes into an endless loop which eventually cause the `ExceedsStackLimit` error. So I changed this to be matching with all interior.